src/framework/core/minizip/ioapi.c

```
@@ -0,0 +1,361 @@
+    ioposix = (FILE_IOPOSIX*)malloc(sizeof(FILE_IOPOSIX));
+    if (ioposix == NULL)
+        return NULL;
+    ioposix->file = file;
+    ioposix->filenameLength = (int)strlen(filename) + 1;
+    ioposix->filename = (char*)malloc(ioposix->filenameLength * sizeof(char));
+    if (ioposix->filename == NULL) {
+        free(ioposix);
+        return NULL;
+    }
+    strncpy((char*)ioposix->filename, filename, ioposix->filenameLength);
+    return (voidpf)ioposix;
```

Cppcheck (2.20.0)

[error] Memory leak

(memleak)